sdk/storage/Azure.Storage.Common/src/StorageTransferStatus.cs

```diff
@@ -12,13 +15,21 @@ public class StorageTransferStatus
         /// <summary>
         /// Bytes transferred.
         /// </summary>
-        public long bytesTransferred { get; set; }
+        public long bytesTransferred { get; internal set;  }
 
         /// <summary>
         /// Number of Files or Blobs tranferred.
         /// </summary>
         public long successfulTransfers { get; internal set; }
 
-        // TODO: failures and skips
+        /// <summary>
+        /// Number of files that failed/
+        /// </summary>
+        public long failedTransfer { get; internal set; }
+
+        /// <summary>
+        /// Number of files that were skipped.
+        /// </summary>
+        public long skippedTransfer { get; internal set; }
```

Comment:
I think we should use separate model for directory upload and single blob upload.

(I'm not exactly sure how this type is used to please correct me if I'm wrong).

Reply:
Yeah that makes sense to do that, for our regular uploads, we just have a regular progress Handler and it just has a `long` which is just the amount of bytes transferred. But not sure if we should make seperate model for that though.

I think this model should be renamed to something like `DirectoryTransferStatus` so single uploads aren't included.